lib/jsdom/browser/Window.js

```diff
@@ -77,34 +78,31 @@ function Window(options) {
   this.clearTimeout = stopTimer.bind(this, window);
   this.__stopAllTimers = stopAllTimers.bind(this, window);
 
+  this._virtualConsole = new VirtualConsole();
+
+  function wrapConsoleMethod(method) {
+    return function () {
+      var args = Array.prototype.slice.call(arguments);
+      window._virtualConsole.emit.apply(window._virtualConsole, [method].concat(args));
+    };
+  }
+
   this.console = {
-    assert: function (expression, message) {
-      if (!expression) {
-        window.raise("error", "Assertion failed: " + message);
-      }
-    },
-    clear: function () {},
-    count: function () {},
-    debug: function () {},
-    error: function (message) {
-      window.raise("error", message);
-    },
-    group: function () {},
-    groupCollapse: function () {},
-    groupEnd: function () {},
-    info: function (message) {
-      window.raise("info", message);
-    },
-    log: function (message) {
-      window.raise("log", message);
-    },
-    table: function () {},
-    time: function () {},
-    timeEnd: function () {},
-    trace: function () {},
-    warn: function (message) {
-      window.raise("warn", message);
-    }
+    assert: wrapConsoleMethod("assert"),
+    clear: wrapConsoleMethod("clear"),
+    count: wrapConsoleMethod("count"),
+    debug: wrapConsoleMethod("debug"),
+    error: wrapConsoleMethod("error"),
+    group: wrapConsoleMethod("group"),
+    groupCollapse: wrapConsoleMethod("groupCollapse"),
+    groupEnd: wrapConsoleMethod("groupEnd"),
+    info: wrapConsoleMethod("info"),
+    log: wrapConsoleMethod("log"),
+    table: wrapConsoleMethod("table"),
+    time: wrapConsoleMethod("time"),
+    timeEnd: wrapConsoleMethod("timeEnd"),
+    trace: wrapConsoleMethod("trace"),
+    warn: wrapConsoleMethod("warn")
```

Comment:
I think it's fine for now. Eventually when I move this stuff off of the instance and on to the prototype I can consider re-organizing.